core-bundle/src/Routing/PageFinder.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\PageModel;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\ExceptionInterface as RoutingExceptionInterface;
+use Symfony\Component\Routing\Matcher\RequestMatcherInterface;
+
+class PageFinder
+{
+    private ContaoFramework $framework;
+    private RequestMatcherInterface $pageRequestMatcher;
+    private RequestMatcherInterface $notFoundRequestMatcher;
+
+    public function __construct(ContaoFramework $framework, RequestMatcherInterface $pageRequestMatcher, RequestMatcherInterface $notFoundRequestMatcher)
+    {
+        $this->framework = $framework;
+        $this->pageRequestMatcher = $pageRequestMatcher;
+        $this->notFoundRequestMatcher = $notFoundRequestMatcher;
+    }
+
+    /**
+     * Find the root page matching the request host and optionally an Accept-Language header.
+     * If $acceptLanguage is not given, it will always return the fallback root page.
+     */
+    public function findRootPageForHost(string $hostname, string $acceptLanguage = null): ?PageModel
+    {
+        $request = Request::create('http://'.$hostname);
+        $request->headers->set('Accept-Language', $acceptLanguage ?? '');
+
+        return $this->matchRootPageForRequest($request);
+    }
+
+    /**
+     * Finds all root pages matching the given host name. It will first look for a matching
+     * root page through routing and then find all root pages with the same tl_page.dns value.
+     *
+     * @return array<PageModel>
+     */
+    public function findAllRootPagesForHost(string $hostname): array
```

Comment:
```suggestion
    public function findRootPagesForHost(string $hostname): array
```
I think the word “all” is not needed here, and may be confusing because you only get published ones, not all.

Reply:
@aschempp this is unresolved so the review makes no sense.

Reply:
changed in 1e2f600e0a4df54a41b195f4a2ed436df394182f